src/js/hooks/useBodyScrollLock.ts

```diff
@@ -0,0 +1,11 @@
+import * as React from 'react';
+
+const useBodyScrollLock = (isInitiallyLocked = false, scrollLockClass = 'scroll-lock') => {
+  const [isBodyScrollLocked, setIsBodyScrollLocked] = React.useState(isInitiallyLocked);
+
+  document.body.classList.toggle(scrollLockClass, isBodyScrollLocked);
+
+  return { isBodyScrollLocked, setIsBodyScrollLocked };
+};
+
+export { useBodyScrollLock };
```

Comment:
This hook is a lean wrapper around adding a scroll lock class to the DOM body, returning the current state and the means to update that state. Useful for improved UX with modal dialogs and this new drawer component.